src/Interpreters/Context.h

```diff
@@ -351,6 +351,11 @@ class Context: public std::enable_shared_from_this<Context>
                                                         /// It's shared with all children contexts.
     MergeTreeTransactionHolder merge_tree_transaction_holder;   /// It will rollback or commit transaction on Context destruction.
 
+    /// For ON CLUSTER queries we check access on initiator node before adding query to the queue
+    /// so we don't need to check it again on queue processing.
+    /// Moreover, it may be not possible to check access on queue processing because we execute query internally, without a user.
```

Comment:
It's not that easy. AFAIK, we can execute RBAC-queries without a user only in `DDLWorker`. But `DDLWorker` never executes a query in global context. It creates its own copy of global context and then uses it to create new query context for each query (as it should be).

Reply:
Ok, let's move `checkGranteeIsAllowed()` to `ContextAccess` where it could check `is_full_access` and thus work properly, no matter it's the global context or its copy or not.